src/CodeGenerator/src/Generator/InputGenerator.php

```diff
@@ -154,8 +154,25 @@ public function generate(Operation $operation): ClassName
                         'NAME' => $member->getName(),
                         'CLASS' => $memberClassName->getName(),
                     ]);
-                } elseif ($mapValueShape instanceof ListShape || $mapValueShape instanceof MapShape) {
-                    throw new \RuntimeException('Recursive ListShape are not yet implemented');
+                } elseif ($mapValueShape instanceof ListShape) {
+                    $listMember = $mapValueShape->getMember();
+                    $listMemberShape = $listMember->getShape();
+                    if (!$listMemberShape instanceof StructureShape) {
+                        throw new \RuntimeException('Recursive ListShape with non StructureShape member is not implemented.');
+                    }
+                    $memberClassName = $this->objectGenerator->generate($listMemberShape);
+                    $constructorBody .= strtr('
+                        $this->NAME = [];
+                        foreach ($input["NAME"] ?? [] as $key => $item) {
+                            $this->NAME[$key] = array_map(function($v) {return CLASS::create($v);}, $item);
+                        }
+                    ', [
+                        'NAME' => $member->getName(),
```

Comment:
`getLocation() ?? getName()`?

Reply:
We don't do that for the section just above the added code. Im not sure why..

Reply:
maybe because it's a map... I'm fine with this